### PR TITLE
Adding rdma support documentation for intel vf's

### PR DESCRIPTION
**Pull request by Eoghan1232**
published docker image available:
RHEL: https://hub.docker.com/r/intel/rdma_rhel_intel

signed off: eoghan.russell@intel.com

### CODE REVIEW

**Comment by SchSeba** [comment]
Hi @Eoghan1232,

thanks for the PR!
question do we need any changes in the device plugin code?
can you please run `env` inside the container I will like to see the mounts we add for the intel RDMA.
can we use the device in that mode for DPDK applications or still the VF needs to be in VFIO driver to work with dpdk?

last question what is the use-case for RDMA with intel? for mlx we need this for DPDK for example.

**Comment by Eoghan1232** [comment]
> Hi @Eoghan1232,
> 
> thanks for the PR! question do we need any changes in the device plugin code? can you please run `env` inside the container I will like to see the mounts we add for the intel RDMA. can we use the device in that mode for DPDK applications or still the VF needs to be in VFIO driver to work with dpdk?
> 
> last question what is the use-case for RDMA with intel? for mlx we need this for DPDK for example.

Hi @SchSeba !
no code changes are needed to the device plugin, it works as is.
sure, here is the output to `env` :
```
root@testpod1:/# env
KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_SERVICE_PORT=443
HOSTNAME=testpod1
PWD=/
PCIDEVICE_INTEL_COM_INTEL_SRIOV_NETDEVICE=0000:5e:01.1
HOME=/root
KUBERNETES_PORT_443_TCP=tcp://10.96.0.1:443
TERM=xterm
SHLVL=1
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_ADDR=10.96.0.1
KUBERNETES_SERVICE_HOST=10.96.0.1
KUBERNETES_PORT=tcp://10.96.0.1:443
KUBERNETES_PORT_443_TCP_PORT=443
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
_=/usr/bin/env
```
This is for enablement for use cases such as kernel based networking (as opposed to user space/DPDK). 
the device would still need to be bound to VFIO in order to use DPDK.

The use case is consistent enablement of hardware offload features available on Intel Network Adapters

**Comment by Eoghan1232** [comment]
> this is weird what version of the device plugin you use? I don't see the new INFO environment with all the mounts

I am currently running an older version of DP on my system, not the latest.
I can redeploy DP as latest and get the output of `env` for you.

**Comment by SchSeba** [comment]
that will be great I will like to see the RDMA mount points that we mount into the pod

**Comment by Eoghan1232** [comment]
@SchSeba 

```
root@testpod2:/# env
KUBERNETES_SERVICE_PORT_HTTPS=443
KUBERNETES_SERVICE_PORT=443
HOSTNAME=testpod2
PWD=/
PCIDEVICE_INTEL_COM_INTEL_SRIOV_NETDEVICE=0000:5e:01.1
HOME=/root
KUBERNETES_PORT_443_TCP=tcp://10.96.0.1:443
TERM=xterm
PCIDEVICE_INTEL_COM_INTEL_SRIOV_NETDEVICE_INFO={"0000:5e:01.1":{"generic":{"deviceID":"0000:5e:01.1"},"rdma":{"rdma_cm":"/dev/infiniband/rdma_cm","uverbs":"/dev/infiniband/uverbs3"}}}
SHLVL=1
KUBERNETES_PORT_443_TCP_PROTO=tcp
KUBERNETES_PORT_443_TCP_ADDR=10.96.0.1
KUBERNETES_SERVICE_HOST=10.96.0.1
KUBERNETES_PORT=tcp://10.96.0.1:443
KUBERNETES_PORT_443_TCP_PORT=443
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
_=/usr/bin/env

```

**Review by adrianchiris — COMMENTED** [review]
a couple of nits otherwise LGTM !

File: docs/rdma/README.md [hunk 1]
```diff
@@ -5,13 +5,16 @@ RDMA supports zero-copy networking by enabling the network adapter to transfer d
...
 * Mellanox ConnectX®-4 Lx Adapter
 * Mellanox ConnectX®-5 Adapter
+* Intel E810-C Adapter
 
 ## RDMA modules:
 * Mellanox ConnectX®-4 Lx, ConnectX®-5 Adapters mlx5_core or mlx5_ib
+* Intel E810-C Adapter ice and iavf
 
 ## Privileges
 IPC_LOCK capability privilege is required for RMA application to function properly in Kubernetes Pod.
```
[the remark this answers is not in the document]
Comment (Eoghan1232) [reply docs/rdma/README.md:20]: thanks for the feedback @adrianchiris , I've addressed this now.

**Review by adrianchiris — APPROVED** [review]

**Review by SchSeba — APPROVED** [review]

**Pull request merged**